hal_st_lwip/instantiations_lwip/CMakeLists.txt

```diff
@@ -15,8 +18,15 @@ target_link_libraries(hal_st.instantiations_lwip PUBLIC
 )
 
 target_sources(hal_st.instantiations_lwip PRIVATE
-    Ethernet.cpp
```

Comment:
I would keep these files here, since emil_build_for already took care of not building these files for other platforms. The upside of leaving in Ethernet.cpp is that the files show up in a Visual Studio solution